roles/tackle/tasks/main.yml

```diff
@@ -519,3 +452,112 @@
+  register: pathfinder_pod
+
+- when:
+    - (pathfinder_pod.resources | length) > 0
+  block:
+    - name: "Wait for Pathfinder to be Ready"
+      kubernetes.core.k8s_info:
+        api_version: v1
+        kind: Pod
+        namespace: "{{ app_namespace }}"
+        label_selectors:
+          - app.kubernetes.io/name = {{ app_name }}-{{ pathfinder_component_name }}
+        wait: true
+        wait_condition:
+          type: "Ready"
+          status: "True"
+        wait_timeout: 240
+
+    - name: "Wait for the Hub to be Ready"
+      kubernetes.core.k8s_info:
+        api_version: v1
+        kind: Pod
+        namespace: "{{ app_namespace }}"
+        label_selectors:
+          - app.kubernetes.io/name = {{ app_name }}-{{ hub_component_name }}
+        wait: true
+        wait_condition:
+          type: "Ready"
+          status: "True"
+        wait_timeout: 240
+
+    - name: Retrieve Hub Secret
+      kubernetes.core.k8s_info:
+        api_version: v1
+        kind: Secret
+        name: "{{ hub_secret_name }}"
+        namespace: "{{ app_namespace }}"
+      register: hub_secret
+
+    - name: Set Hub Token
+      set_fact:
+        hub_key: "{{ hub_secret.resources[0].data.addon_token | b64decode }}"
+
+    - name: Migrate assessments
+      shell: |
+        /usr/local/bin/migrate-pathfinder-assessments.py \
+        -p http://{{ pathfinder_service_name }}:8080/pathfinder \
+        -b http://{{ hub_service_name }}:8080 \
+        -t $(/usr/local/bin/jwt.sh {{ hub_key }})
+      changed_when: false
+
+    - name: "Remove PathFinder PostgreSQL Secret"
+      k8s:
+        state: absent
+        api_version: v1
+        kind: Secret
+        name: "{{ pathfinder_database_secret_name }}"
+        namespace: "{{ app_namespace }}"
+
+    - name: "Remove PathFinder PostgreSQL Service"
+      k8s:
+        state: absent
+        api_version: v1
+        kind: Service
+        name: "{{ pathfinder_database_service_name }}"
+        namespace: "{{ app_namespace }}"
+
+    - name: "Remove PathFinder PostgreSQL Deployment"
+      k8s:
+        state: absent
+        api_version: apps/v1
+        kind: Deployment
+        name: "{{ pathfinder_database_deployment_name }}"
+        namespace: "{{ app_namespace }}"
+
+    - name: "Remove PathFinder Service"
+      k8s:
+        state: absent
+        api_version: v1
+        kind: Service
+        name: "{{ pathfinder_service_name }}"
+        namespace: "{{ app_namespace }}"
+
+    - name: "Remove PathFinder Deployment"
+      k8s:
+        state: absent
+        api_version: apps/v1
+        kind: Deployment
+        name: "{{ pathfinder_deployment_name }}"
+        namespace: "{{ app_namespace }}"
+
+- name: "Remove PathFinder PostgreSQL Volume"
+  k8s:
+    state: absent
+    api_version: v1
+    kind: PersistentVolumeClaim
+    name: "{{ pathfinder_database_data_volume_claim_name }}"
+    namespace: "{{ app_namespace }}"
+  when:
+    - (pathfinder_delete_db_volume|bool)
```

Comment:
I put the DB volume removal behind a flag since there's no going back. To me it seems like this should be a conscious users decision to make once they feel confident everything is as it should be. They can also delete it manually `oc delete pvc ...`.